frontend/src/pages/modelRegistrySettings/CreateModal.tsx

```diff
@@ -245,16 +279,27 @@ const CreateModal: React.FC<CreateModalProps> = ({ onClose, refresh }) => {
             )}
           </FormGroup>
           <FormGroup label="Password" isRequired fieldId="mr-password">
-            <PasswordInput
-              isRequired
-              type={showPassword ? 'text' : 'password'}
-              id="mr-password"
-              name="mr-password"
-              value={password}
-              onBlur={() => setIsPasswordTouched(true)}
-              onChange={(_e, value) => setPassword(value)}
-              validated={isPasswordTouched && !hasContent(password) ? 'error' : 'default'}
-            />
+            {mr ? (
+              <ModelRegistryDatabasePassword
+                password={password || ''}
+                setPassword={setPassword}
+                isPasswordTouched={isPasswordTouched}
+                setIsPasswordTouched={setIsPasswordTouched}
+                showPassword={showPassword}
+                editRegistry={mr}
+              />
+            ) : (
+              <PasswordInput
+                isRequired
+                type={showPassword ? 'text' : 'password'}
+                id="mr-password"
+                name="mr-password"
+                value={password}
+                onBlur={() => setIsPasswordTouched(true)}
+                onChange={(_e, value) => setPassword(value)}
+                validated={isPasswordTouched && !hasContent(password) ? 'error' : 'default'}
```

Comment:
This is the same set of props that gets rendered inside ModelRegistryDatabasePassword. Can we just make the `editRegistry` prop of that component optional, and have its editing-specific behavior be conditional?